### ISSUE TITLE
[BUG] preversion script has access to new version

### ISSUE THREAD

**Issue by erkstruwe** [npm/cli#3157]
### Current Behavior:
When running `npm version` and the `preversion` script is executed, the value of the `npm_package_version` environment variable holds the *new* version.

### Expected Behavior:
> Run the `preversion` script. These scripts have access to the *old* `version` in package.json.
> - https://docs.npmjs.com/cli/v7/commands/npm-version#description (emphasis added by me)

### Steps To Reproduce:
1. Create new package with this `package.json`:
```
{
  "name": "test",
  "version": "1.0.0",
  "scripts": {
    "preversion": "cross-var echo \"preversion $npm_package_version\"",
    "version": "cross-var echo \"version $npm_package_version\"",
    "postversion": "cross-var echo \"postversion $npm_package_version\""
  },
  "devDependencies": {
    "cross-var": "^1.1.0"
  }
}
```
2. Run `npm install`
3. Run `npm version patch`
4. Output is
```
preversion 1.0.1
version 1.0.1
postversion 1.0.1
```

### Environment:
- OS: Windows 10
- Node: 14.16.1 // edited. original post said 16.0.0
- npm: 6.14.12 // edited. orignal post said 7.10.0

**Comment by ljharb** [comment]
I think this is the correct behavior in preversion - you can read the old version from package.json, but the new one is nowhere else besides the env. In other words, i think maybe the docs are incorrect here.

**Comment by ljharb** [comment]
I would expect the env to hold the new version in preversion, and the old version in postversion, since package.json has the “other” piece of information in both cases.

**Comment by erkstruwe** [comment]
I like the idea.

Since reading `package.json` from command line is quite cumbersome, the ideal solution IMHO would be a new set of environment variables only available in `version`, `preversion`, and `postversion` scripts. Something along the lines of `npm_package_previous_version` and `npm_package_next_version` or `npm_package_version_old` and `npm_package_version_new`, depending on whatever the naming convention for those variables is.

**Comment by ljharb** [comment]
`node -pe “require(\”./package.json\”).version` doesn't seem super cumbersome.

**Comment by isaacs** [comment]
Here's what I'm seeing in the latest and greatest npm version 7:

```
$ npm version major

> preversion
> env | grep npm_ | grep version | sort

npm_command=version
npm_lifecycle_event=preversion
npm_lifecycle_script=env | grep npm_ | grep version | sort
npm_new_version=2.0.0
npm_old_version=1.0.0
npm_package_version=1.0.0

> version
> env | grep npm_ | grep version | sort

npm_command=version
npm_lifecycle_event=version
npm_lifecycle_script=env | grep npm_ | grep version | sort
npm_new_version=2.0.0
npm_old_version=1.0.0
npm_package_version=2.0.0

> postversion
> env | grep npm_ | grep version | sort

npm_command=version
npm_lifecycle_event=postversion
npm_lifecycle_script=env | grep npm_ | grep version | sort
npm_new_version=2.0.0
npm_old_version=1.0.0
npm_package_version=2.0.0
v2.0.0
```

`preversion` has `npm_package_version` set to the old version, `version` and `postversion` have it set to the new version.  In all three `npm_new_version` and `npm_old_version` are set as expected.

So it looks like this is fixed, or there's something else going on here?

**Comment by erkstruwe** [comment]
@isaacs You're right.

I used the wrong executables in my tests. I'm sorry. I can confirm the bug shows in npm 6.14.12, but is fixed in npm 7.10.0.

Thanks for your support, guys. We might close this...
